core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -682,171 +670,90 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         });
     }
 
-    static class ProcessClusterState {
-        final ClusterState clusterState;
-        volatile boolean processed;
-
-        ProcessClusterState(ClusterState clusterState) {
-            this.clusterState = clusterState;
-        }
-    }
+    void handleNewClusterStateFromMaster(String reason) {
```

Comment:
I agrees the current name carries less meaning in the way it works now. applyLastCommitedClusterState is not ideal in my mind as we don't know which state will be selected. that's up to the queue. How about - processNextPendingClusterState? I'll use this if there is no objection...